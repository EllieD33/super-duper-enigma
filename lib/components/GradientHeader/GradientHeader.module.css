.pinkToBlue {
    background: linear-gradient(
        107deg,
        rgba(252, 11, 207, 1) 0%,
        rgba(0, 189, 255, 1) 100%
    );
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;

    /* For Firefox */
    background-clip: text;
    color: transparent;
}

.pink {
    background: linear-gradient(
        90deg,
        rgb(252, 11, 207) 0%,
        rgba(99, 9, 83, 1) 79%
    );
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;

    /* For Firefox */
    background-clip: text;
    color: transparent;
}

.blue {
    background: #00bdff;
    background: linear-gradient(
        90deg,
        rgba(0, 189, 255, 1) 0%,
        rgba(11, 81, 107, 1) 79%
    );
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;

    /* For Firefox */
    background-clip: text;
    color: transparent;
}
