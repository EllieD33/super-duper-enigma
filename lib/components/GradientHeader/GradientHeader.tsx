import React, { ReactElement } from "react";
import styles from "./GradientHeader.module.css";
import Text, { TextProps } from "../Text/Text";

export interface GradientHeaderProps extends TextProps {
    variant?: "pinkToBlue" | "pink" | "blue";
}

const GradientHeader = ({
    variant = "pinkToBlue",
    children,
    ...props
}: GradientHeaderProps): ReactElement => {
    return (
        <Text className={styles[variant]} {...props}>
            {children}
        </Text>
    );
};

export default GradientHeader;
