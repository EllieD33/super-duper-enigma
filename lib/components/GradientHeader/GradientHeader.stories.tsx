import type { Meta, StoryObj } from "@storybook/react";
import GradientHeader, { GradientHeaderProps } from "./GradientHeader";

const meta = {
    title: "Library/GradientHeader/GradientHeader",
    component: GradientHeader,
    parameters: {
        layout: "centered",
    },
    tags: ["autodocs"],
    argTypes: {},
} satisfies Meta<typeof GradientHeader>;

export default meta;
type Story = StoryObj<typeof meta>;

const defaultProps: GradientHeaderProps = {
    children: "Experts in Web3 Security & Crisis Response",
    styleAs: "Heading1",
};

export const Default: Story = {
    args: {
        ...defaultProps,
    },
};
