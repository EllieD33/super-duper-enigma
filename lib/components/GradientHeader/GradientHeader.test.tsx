import React from "react";
import { render, screen } from "@testing-library/react";
import GradientHeader, { GradientHeaderProps } from "./GradientHeader";

const defaultProps: GradientHeaderProps = {
    children: "Experts in Web3 Security & Crisis Response",
    styleAs: "Heading1",
};

describe("GradientHeader", () => {
    it("should render the default component", () => {
        render(<GradientHeader {...defaultProps} />);
        const { container } = render(<GradientHeader {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });
});
