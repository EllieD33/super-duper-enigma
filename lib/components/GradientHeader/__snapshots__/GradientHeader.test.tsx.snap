// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientHeader should render the default component 1`] = `
<div>
  <p
    class="left pinkToBlue"
    style="color: rgb(51, 51, 51); font-weight: 700; font-size: 2rem; font-family: 'Roboto', sans-serif;"
  >
    Experts in Web3 Security & Crisis Response
  </p>
</div>
`;
